src/compiler.ts

```diff
@@ -8612,6 +8621,9 @@ export class Compiler extends DiagnosticEmitter {
         varTypes,
         module.flatten(stmts, sizeTypeRef)
       );
+      if (!this.options.willOptimize) {
+        module.setLocalNames(funcRef, instance.localsByIndex.map<string>(local => local.name));
+      }
```

Comment:
While all this pattern the same I guess make sense refactor `setLocalNames`'s signature to this:
```ts
  setLocalNames(funcRef: FunctionRef, locals: Local[] | null = null): void {
    if (locals == null) return;
    let localNames = locals.map<string>(local => local.name);
    ...
  }
```
and call sites simplifies to:
```ts
     // "locals" already should defined as "instance.localsByIndex" above
      if (!this.options.willOptimize) {
        module.setLocalNames(funcRef, locals);
      }
```

Reply:
I don't want to import `Local` into `Module.ts` because it add complexity and in my opinion Module function should be very simple for easier using. I will extract the `Local[]` -> `string[]` logic into a helper function placed in `compiler.ts`